Sources/Protocols/Designable/GradientDesignable.swift

```diff
@@ -6,145 +6,153 @@
 
-fileprivate extension GradientDesignable where Self: UIView {
-  func configureGradient(startColor: UIColor, endColor: UIColor) {
-
-    let gradientLayer = makeGradientLayer()
-    gradientLayer.colors = [startColor.cgColor, endColor.cgColor]
-    gradientLayer.configurePoints(with: startPoint)
+extension GradientDesignable where Self: UIView {
 
+  public func configureGradient() {
     subviews.filter { $0 is PrivateGradientView }.forEach {
       $0.removeFromSuperview()
     }
 
-    let gradientView = PrivateGradientView(frame: bounds, gradientLayer: gradientLayer)
-    insertSubview(gradientView, at: 0)
-  }
+    guard let gradientLayer = makeGradientLayer() else {
+      return
+    }
 
-  func makeGradientLayer() -> CAGradientLayer {
-    let gradientLayer: CAGradientLayer = CAGradientLayer()
-    gradientLayer.frame = bounds
     gradientLayer.cornerRadius = layer.cornerRadius
-    return gradientLayer
+    gradientLayer.frame = bounds
+    let gradientView = PrivateGradientView(frame: bounds, gradientLayer: gradientLayer)
+    insertSubview(gradientView, at: 0)
   }
 
-  func configurePredefinedGradient() -> GradientColor? {
-    guard let gradientType = predefinedGradient else {
-      return nil
-    }
-    return gradientType.colors
-  }
 }
 
-public extension GradientDesignable where Self: UINavigationBar {
-  public func configureGradient() {
-    let predefinedGradient = configurePredefinedGradient()
-    if let startColor = startColor, let endColor = endColor {
-      configureGradient(startColor: startColor, endColor: endColor)
-    } else if let startColor = predefinedGradient?.start, let endColor = predefinedGradient?.end {
-      configureGradient(startColor: startColor, endColor: endColor)
-    }
-  }
+// MARK: - NavigationBar
 
-  fileprivate  func configureGradient(startColor: UIColor, endColor: UIColor) {
+extension GradientDesignable where Self: UINavigationBar {
+
+  public func configureGradient() {
+    let navBarFrame = CGRect(x: 0, y: 0, width: bounds.width, height: UIApplication.shared.statusBarFrame.height)
     let gradientLayer = makeGradientLayer()
-    gradientLayer.colors = [startColor.cgColor, endColor.cgColor]
-    gradientLayer.configurePoints(with: startPoint)
+    gradientLayer?.frame = navBarFrame
+    gradientLayer?.cornerRadius = layer.cornerRadius
 
-    let image = gradientLayer.makeImage()
+    let image = gradientLayer?.makeImage()
     setBackgroundImage(image, for: .default)
   }
 
-  fileprivate func makeGradientLayer() -> CAGradientLayer {
-    let gradientLayer: CAGradientLayer = CAGradientLayer()
+}
 
-    let statusBarHeight: CGFloat = 64 // UIApplication.shared.statusBarFrame.height
-    let navBarFrame = CGRect(x: 0, y: 0, width: bounds.width, height: statusBarHeight)
-    gradientLayer.frame = navBarFrame
+// MARK: - Layers
 
-    gradientLayer.cornerRadius = layer.cornerRadius
-    return gradientLayer
+extension CALayer {
+
+  fileprivate func makeImage() -> UIImage? {
```

Comment:
I think we can use `private` now

Reply:
Can't do that, it's still in use in the `GradientDesignable` extension used for the navigation bar, otherwise, I could move it, but it's quite useful to have it in the layer extension 😬